daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/HexBinaryLengthParsers.scala

```diff
@@ -34,11 +34,12 @@ sealed abstract class HexBinaryLengthParser(override val context: ElementRuntime
     val dis = start.dataInputStream
     val currentElement = start.simpleElement
     val nBits = getLengthInBits(start)
-    if (nBits == 0) {
+    val nBytes = (nBits + 7) / 8
+    if (nBytes == 0) {
       currentElement.setDataValue(zeroLengthArray)
-    } else if (nBits > Int.MaxValue) {
+    } else if (nBytes > Int.MaxValue) {
       // Integer overflow will occurr, recommend using blob instead of hexBinary
-      PE(start, "Data is too large for xs:hexBinary. Consider using blobs instead: https://s.apache.org/daffodil-blob-feature")
+      PE(start, "Length for xs:hexBinary exceeds maximum of %s bytes: %s", Int.MaxValue, nBytes)
```

Comment:
A good reason to have a tunable size here is so you can crank it way down for testing purposes and cover all these error cases. People will really run into them with large data, so we really should cover them.

Reply:
We do have a tunable to configure the blob size, and some test that crank it down to zero. So we should be able to easily test the ``hex binary > chunk size`` limit. I can add a tunable for this and default it to MaxInt.